### PR TITLE
test/crimson: add a test to verify that errorator does not copy 

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by athanatos — APPROVED** [review]

File: src/test/crimson/test_errorator.cc [hunk 1]
```diff
@@ -6,9 +6,9 @@
 #include "crimson/common/errorator.h"
 #include "crimson/common/log.h"
 
-struct do_until_test_t : public seastar_test_suite_t {
```
Comment (rzarzynski) [inline src/test/crimson/test_errorator.cc:9]: nit: typo in the commit message:

`so we can reuse this test suite for more erroratoer related tests.` -> `so we can reuse this test suite for more errorator related tests.`

File: src/test/crimson/test_errorator.cc [hunk 2]
```diff
@@ -18,6 +18,23 @@ struct errorator_test_t : public seastar_test_suite_t {
       }
     });
   }
+  struct noncopyable_t {
```
Comment (rzarzynski) [inline src/test/crimson/test_errorator.cc:21]: How about [`boost::noncopyable`](https://www.boost.org/doc/libs/1_65_0/libs/core/doc/html/core/noncopyable.html)?

**Review by rzarzynski — APPROVED** [review]
Just a nit / a suggestion. Overall LGTM. More unittesting for errorator is a wonderful idea. Going to follow.

File: src/test/crimson/test_errorator.cc (hunk 2 above)
Comment (tchaikov) [reply src/test/crimson/test_errorator.cc:21]: actually, i was using `boost::noncopyable`, but after a second thought, still think that the homebrew version is more straightforward.

File: src/test/crimson/test_errorator.cc (hunk 1 above)
Comment (tchaikov) [reply src/test/crimson/test_errorator.cc:9]: thanks. fixed and repushed.

**Comment by tchaikov** [comment]
> Just a nit / a suggestion. Overall LGTM. More unittesting for errorator is a wonderful idea. Going to follow.

yeah, when reviewing Sam's PR. i was wondering if 
```c++
return create_a_heavy_weight_object().safe_then([](auto the_heavy_weight_object) {
  //...
}
```
and
```c++
return create_a_heavy_weight_object().safe_then([](auto&& the_heavy_weight_object) {
  //...
}
```
are different. and if the former calls the copy constructor of `heavy_weight_object`.

and then created this test to verify my analysis. also, i think a "runnable document" would help also.

**Pull request merged**